urllib3/_backends/__init__.py

```diff
@@ -1,9 +1,56 @@
+import logging
+
 from urllib3.packages import six
-from .sync_backend import SyncBackend
 
-__all__ = ['SyncBackend']
 
-if six.PY3:
+def load_trio_backend():
+    if not six.PY3:
+        raise ValueError("trio backend requires Python 3")
     from .trio_backend import TrioBackend
+    return TrioBackend
+
+
+def load_twisted_backend():
+    if not six.PY3:
+        raise ValueError("twisted backend requires Python 3")
     from .twisted_backend import TwistedBackend
-    __all__ += ['TrioBackend', 'TwistedBackend']
+    return TwistedBackend
+
+
+def load_sync_backend():
+    from .sync_backend import SyncBackend
+    return SyncBackend
+
+
+BACKENDS = {
+    "sync": load_sync_backend,
+    "trio": load_trio_backend,
+    "twisted": load_twisted_backend,
+}
+
+
+def available_backends():
+    """
+    Returns a list of backend names that we are able to successfully import.
+    """
+    output = []
+    for backend_name, load_backend_fn in sorted(BACKENDS.items()):
+        try:
+            load_backend_fn()
+            output.append(backend_name)
+        except:  # noqa
```

Comment:
You probably want at least `except Exception` in order to avoid catching KeyboardInterrupt exceptions.

Any good reason not to simply use `ImportError`? I guess the import can fail with different errors, but as a user, I would like to know if a dependency is half-installed.

Am I missing something?

Reply:
That sounds like a good idea - thanks!